test/unstable/dmapped-sparse.chpl

```diff
@@ -7,4 +7,16 @@ writeln(SD);
 var DD = D dmapped DefaultDist();   // warning: 'dmapped' is unstable
 writeln(DD);
 
-compilerError("done");
+var dm = new dmap(new DefaultDist());
+testit(dm);
+
+proc testit(arg: _distribution) {
+  compilerError("done");
+}
+
+// NB this overload does not match the above call testit(dm)
+// because by resolution time dm's dmap has been converted to '_distribution'
+// while this overload's dmap is still 'dmap'.
+proc testit(arg: dmap) {
+  compilerError("arg: dmap");
+}
```

Comment:
I'm not sure I understand why this is needed?

Does this not trigger the unstable warning?
```chapel
var dm = new dmap(new DefaultDist());
writeln(dm);
```

Reply:
@jabraham17 Good question! Your simple code triggers the unstable warning as expected, on the first line.

The reason I added the second overload is to illustrate an implementation deficiency where in some cases we cannot use `dmap` and instead need to use `_distribution`. It is a problem because _distribution is not user facing. I am not pursuing this further because we want to eliminate `dmap` and `_distribution` altogether, see ex. #22657.